lib/teiserver/battle/balance/default_balance.ex

```diff
@@ -0,0 +1,79 @@
+defmodule Teiserver.Battle.Balance.DefaultBalance do
+  @moduledoc """
+  This will call other balancers depending on circumstances
+  """
+  alias Teiserver.Battle.Balance.SplitNoobs
+  alias Teiserver.Battle.Balance.LoserPicks
+  alias Teiserver.Battle.Balance.SplitNoobsTypes, as: SN
+  alias Teiserver.Battle.Balance.BalanceTypes, as: BT
+
+  @doc """
+  Main entry point used by balance_lib
+  """
+  @spec perform([BT.expanded_group()], non_neg_integer(), list()) :: any()
+  def perform(expanded_group, team_count, opts \\ []) do
+    case should_use_algo(expanded_group, team_count) do
+      "split_noobs" ->
+        SplitNoobs.perform(expanded_group, team_count, opts)
+
+      _ ->
+        LoserPicks.perform(expanded_group, team_count, opts)
+    end
+  end
+
+  @spec should_use_algo([BT.expanded_group()], integer()) ::
+          String.t()
+  def should_use_algo(expanded_group, team_count) do
```

Comment:
I would refactor this function into something like `balance_module`, and instead of returning a string, you can directly return the module. Then the caller can do something like
```elixir
balance_module(expanded_group, team_count, opts).perform(expanded_group, team_count, opts)
```

This is a nice approach to define an interface. If you feel like it you can also define a [behaviour](https://hexdocs.pm/elixir/typespecs.html#behaviours) for the balance modules.